### PR TITLE
created company, vacancy and candidate CRUD Operation

### PR DESCRIPTION
**Pull request by AhmedAbdullaal**

### CODE REVIEW

File: back-end/src/main/java/com/example/demo/rest/dto/VacancyDto.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import com.example.demo.rest.dto.common.RestDto;
+import lombok.Data;
+import java.time.LocalDateTime;
+
+@Data
+public class VacancyDto extends RestDto {
+
+
+    private Integer vacancies;
```
Comment (mhassanInfo90) [inline back-end/src/main/java/com/example/demo/rest/dto/VacancyDto.java:11]: ايه vacancies ؟؟

File: back-end/src/main/java/com/example/demo/entity/Vacancy.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+@EqualsAndHashCode(of = "id", callSuper = false)
+@ToString(of = "id", callSuper = false)
+public class Vacancy extends JPAEntity {
+
+    @Column(name = "vacancies")
+    private Integer vacancies;
+
+    @Column(name = "post_by", updatable = false)
+    @CreatedBy
+    private String postBy;
```
Comment (mhassanInfo90) [inline back-end/src/main/java/com/example/demo/entity/Vacancy.java:24]: you use Created By this mean Hold user name login user ? if yes you need to create users table also to make sue insert data will valid

File: back-end/src/main/java/com/example/demo/rest/dto/VacancyDto.java (hunk 1 above)
Comment (AhmedAbdullaal) [reply back-end/src/main/java/com/example/demo/rest/dto/VacancyDto.java:11]: عدد الوظائف الخالية في ال position

File: back-end/src/main/java/com/example/demo/entity/Vacancy.java (hunk 2 above)
Comment (AhmedAbdullaal) [reply back-end/src/main/java/com/example/demo/entity/Vacancy.java:24]: I mean by postBy the employee that made this vacancy

File: back-end/src/main/java/com/example/demo/entity/Candidate.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+@ToString(of = "id", callSuper = false)
+public class Candidate extends JPAEntity {
+
+    @Column(name = "name")
+    private String name;
+
+    @Column(name = "phone")
+    private String phone;
+
+    @Column(name = "E-mail")
```
Comment (mhassanInfo90) [inline back-end/src/main/java/com/example/demo/entity/Candidate.java:22]: Email like E-Email Working !!

File: back-end/src/main/java/com/example/demo/entity/Vacancy.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import java.time.LocalDateTime;
+
+@Data
+@Entity
+@EqualsAndHashCode(of = "id", callSuper = false)
+@ToString(of = "id", callSuper = false)
+public class Vacancy extends JPAEntity {
+
+    @Column(name = "vacancies")
+    private Integer vacancies;
```
Comment (mhassanInfo90) [inline back-end/src/main/java/com/example/demo/entity/Vacancy.java:20]: Rename it According to count of number vacancy open in this vacancy

File: back-end/src/main/java/com/example/demo/rest/dto/VacancyDto.java (hunk 1 above)
Comment (mhassanInfo90) [reply back-end/src/main/java/com/example/demo/rest/dto/VacancyDto.java:11]: rename it to meet the meaning

**Pull request merged**